cli/data.go

```diff
@@ -0,0 +1,194 @@
+package cli
+
+import (
+	"bufio"
+	"bytes"
+	"compress/gzip"
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/pkg/errors"
+	datapb "go.viam.com/api/app/data/v1"
+	rdkutils "go.viam.com/rdk/utils"
+	"google.golang.org/protobuf/encoding/protojson"
+	"io"
+	"os"
+	"path/filepath"
+	"strconv"
+)
+
+const (
+	dataDir     = "data"
+	metadataDir = "metadata"
+)
+
+// BinaryData writes the requested data to the passed directory.
+func (c *AppClient) BinaryData(dst string, filter *datapb.Filter) error {
+	fmt.Println(filter.OrgIds)
+
+	if err := c.ensureLoggedIn(); err != nil {
+		return err
+	}
+
+	// TODO: Should we use more limited perms?
+	// TODO: Probably shouldn't re-download files we already have? Maybe we actually should have an exlclude_ids field
+	if err := os.MkdirAll(filepath.Join(dst, dataDir), os.ModePerm); err != nil {
+		return err
+	}
+	if err := os.MkdirAll(filepath.Join(dst, metadataDir), os.ModePerm); err != nil {
+		return err
+	}
+
+	skip := int64(0)
+	for {
+		resp, err := c.dataClient.BinaryDataByFilter(context.Background(), &datapb.BinaryDataByFilterRequest{
+			DataRequest: &datapb.DataRequest{
+				Filter: filter,
+				Skip:   skip,
+				Limit:  1,
+			},
+			IncludeBinary: true,
+			CountOnly:     false,
+		})
+		// TODO: Make sure EOF is properly interpreted. Iirc rpc errors aren't properly parsed by errors.Is.
```

Comment:
Good todo - checking on testing this before merging.

Reply:
Forgot this was a unary rpc, so EOF isn't relevant. Right now we just return an empty response when skip > the amount of data (so when we're done iterating through all the data with skip/limit). Posted in the team channel to see if there's an appropriate status code for this case, but handling it just by checking if the response is empty for now